app/controllers/api/v1/sessions_controller.rb

```diff
@@ -43,6 +43,12 @@ def session_params
       def sign_in(user)
         session[:user_id] = user.id
       end
+
+      # Signs Out the user
+      def sign_out
+        session[:user_id] = nil
+        @current_user = nil
```

Comment:
You are right - the behaviour does not change when it is removed.

Is it me or is there some Rails magic going on?
Shouldn't the right side be ignored if @current_user isn't nil?

![image](https://user-images.githubusercontent.com/43917914/161144156-dafd71e4-85ed-4977-ac57-e065f5d5a833.png)

Reply:
@scouillard I think I get the reason behind your confusing.
The `@current_user` is an instance variable and it stays alive only for the whole `request/response` lifecycle meaning that with each new request a new instance variable will get created.
Upon initial interaction and since `@current_user` will be `nil` the instruction `User.find_by id:...` will get executed fetching the current user data from the db according to its stored id in the session hash.
After this and until a response is issued back the instance variable `@current_user` will always hold that value.
Once a response is issued back Rails garbage collects all created instances for that request and all of the controller instances with their properties will get lost.
Upon the next request the whole process will repeat meaning after a successful signout operation which resets the session data for the client the upcoming requests will send back no user id and each controller instance will have a `nil` `@current_user`.

Reply:
Thank you Amir.
I wasn't thinking in terms of request/response.